docs/dev/modules.md

```diff
@@ -0,0 +1,105 @@
+# Cirq modules 
+
+Cirq has a modular architecture and is organized in a monorepo, all of the modules are in the same folder structure.
+Each module is structured as follows: 
+
+```
+cirq-<module-name>
+   <top level package 1> 
+   <top level package 2>
+   ...
+   setup.py
+   setup.cfg
+   requirements.txt
+   LICENSE
+   README.rst 
+...
+setup.py # metapackage 
+```
+
+Note that typically there is only a single top level package, but there might be exceptions. 
+
+The highest level module, `cirq` is an exception, as it is a metapackage, kind of a "parent" module, that only contains the set of default submodules as requirements. 
```

Comment:
Hmm I don't know about the general organization of this explanation. I'd say: all modules have this structure with cirq-name/cirq_name

Additionally, there is a metapackage "cirq" that's a completely different beast and just depends on the modules.

i.e. don't even bother calling the cirq metapackage a module, highest level module, or parent module.

Reply:
👍